### PR TITLE
Add the first example project

### PR DESCRIPTION
**Pull request by trustin**
- Added the first example `spring-boot-tomcat` as requested by many people
- Updated Checkstyle rules and suppressions to make sure the examples do not have a copyright header because they belong to the public domain.
- Miscellaneous:
  - Added the constructor that accepts `Iterable<? extends HealthChecker>` to `HttpHealthCheckService`

### CODE REVIEW

File: core/src/main/java/com/linecorp/armeria/server/healthcheck/HttpHealthCheckService.java [hunk 1]
```diff
@@ -91,7 +93,16 @@
...
+        this(ImmutableList.copyOf(requireNonNull(healthCheckers, "healthCheckers")));
+    }
+
+    /**
+     * Creates a new instance.
+     *
+     * @param healthCheckers the additional {@link HealthChecker}s
+     */
+    public HttpHealthCheckService(Iterable<HealthChecker> healthCheckers) {
+        this.healthCheckers = ImmutableList.copyOf(requireNonNull(healthCheckers, "healthCheckers"));
```
Comment (trustin) [inline core/src/main/java/com/linecorp/armeria/server/healthcheck/HttpHealthCheckService.java]: Not related with this PR but seemed pretty obvious.

File: examples/spring-boot-tomcat/src/main/java/example/springframework/boot/tomcat/HelloConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+public class HelloConfiguration {
+
+    /**
+     * Extracts a Tomcat {@link Connector} from Spring webapp context.
+     */
+    public static Connector getConnector(ServletWebServerApplicationContext applicationContext) {
+        final TomcatWebServer container =
+                (TomcatWebServer) applicationContext.getWebServer();
+        try {
+            final Field serviceConnectorsField = TomcatWebServer.class.getDeclaredField("serviceConnectors");
```
Comment (imasahiro) [inline examples/spring-boot-tomcat/src/main/java/example/springframework/boot/tomcat/HelloConfiguration.java]: You can simplify this reflection using following code.
```java
        TomcatWebServer container =
                (TomcatWebServer) applicationContext.getWebServer();
        // Starts the container explicitly to make sure all connectors are write back from serviceConnectors.
        container.start();
        Connector tomcatConnector = container.getTomcat().getConnector();
```

Comment (trustin) [reply examples/spring-boot-tomcat/src/main/java/example/springframework/boot/tomcat/HelloConfiguration.java]: w00t! Thanks a lot. :+1:

File: examples/spring-boot-tomcat/build.gradle [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+apply plugin: 'org.springframework.boot'
+
+springBoot {
+    mainClassName = 'example.springframework.boot.tomcat.Main'
+}
+
+dependencies {
+    compile project(':core')
+    compile project(':tomcat')
```
Comment (imasahiro) [inline examples/spring-boot-tomcat/build.gradle]: better to reorder by alphabetical order

**Comment by trustin** [comment]
Thanks, reviewers :-)

**Pull request merged**